geotrek/core/management/commands/loadpaths.py

```diff
@@ -0,0 +1,94 @@
+                            help="Name of the name's attribute inside the file")
+        parser.add_argument('--comments-attribute', '-c', nargs='*', action='store', dest='comment',
+                            help="")
+        parser.add_argument('--encoding', '-e', action='store', dest='encoding', default='utf-8',
+                            help='File encoding, default utf-8')
+        parser.add_argument('--srid', '-s', action='store', dest='srid', default=4326,
+                            help="File's SRID")
+        parser.add_argument('--intersect', '-i', action='store_true', dest='intersect', default=False,
+                            help="Check paths intersect spatial extent and not only within")
+        parser.add_argument('--fail', '-f', action='store_true', dest='fail', default=False,
+                            help="Allows to grant fails")
+
+    def handle(self, *args, **options):
+        verbosity = options.get('verbosity')
+        encoding = options.get('encoding')
+        file_path = options.get('file_path')
+        structure = options.get('structure')
+        name_column = options.get('name')
+        srid = options.get('srid')
+        do_intersect = options.get('intersect')
+        comments_column = options.get('comments')
+        fail = options.get('fail')
+
+        if structure:
+            try:
+                structure = Structure.objects.get(name=structure)
+            except Structure.DoesNotExist:
+                raise CommandError("Structure does not match with instance's structures\n"
+                                   "Change your option --structure")
+        elif Structure.objects.count() < 1:
+            structure = Structure.objects.first()
+        else:
+            raise CommandError("There are 2 structures and you didn't define the option structure\n"
+                               "Use --structure to define it")
+        if verbosity > 0:
+            self.stdout.write("All paths in DataSource will be linked to the structure : %s" % structure)
+
+        ds = DataSource(file_path, encoding=encoding)
+
+        bbox = Polygon.from_bbox(settings.SPATIAL_EXTENT)
+        bbox.srid = settings.SRID
+
+        for layer in ds:
+            for feat in layer:
+                name = feat.get(name_column) if name_column in layer.fields else ''
+                comments = feat.get(comments_column) if comments_column in layer.fields else ''
+                try:
+                    geom = feat.geom.geos
+                    if not isinstance(geom, LineString):
+                        if verbosity > 0:
+                            self.stdout.write("%s's geometry is not a Linestring" % feat)
+                        break
+                    self.check_srid(srid, geom)
+                    geom.dim = 2
+                    if do_intersect and bbox.intersects(geom) or not do_intersect and geom.within(bbox):
+                        try:
```

Comment:
ce trey/except devrait etre beaucoup plus large, juste apres for feat in layer. ca evitera des crash en plein import

Reply:
Ca va quand meme les creer non  dans tous les cas.

Reply:
non